sway-lib-std/src/vec.sw

```diff
@@ -6,16 +6,28 @@ use ::intrinsics::size_of;
 use ::mem::{copy, read, write};
 use ::option::Option;
 
+fn raw_ptr_from_u64(ptr: u64) -> raw_ptr {
+    asm(r1: ptr) {
+        r1: raw_ptr
+    }
+}
+
+fn raw_ptr_into_u64(ptr: raw_ptr) -> u64 {
+    asm(r1: ptr) {
+        r1: u64
+    }
+}
+
 struct RawVec<T> {
-    ptr: u64,
+    ptr: raw_ptr,
```

Comment:
This is going to break the input `Vec` support in the SDks. We should communicate that asap with new issues. We clearly don't have any SDKs tests that use `Vec` as an input yet which is why CI is passing here :) I opened [this issue ](https://github.com/FuelLabs/sway/issues/2922)recently and I hope to get to it soon.

Reply:
Actually I've also opened https://github.com/FuelLabs/fuels-rs/issues/600 which might be enough.